src/ddspdrum/module.py

```diff
@@ -58,6 +62,94 @@ def control_to_sample_rate(self, control: np.array) -> np.array:
             )
             return resample(control, num_samples)
 
+    def add_parameter(self, parameter_id: str, value: float, minimum: float,
+                      maximum: float, scale: float = 1):
+        """
+        Add a new parameter to this module.
+
+        Parameters
+        ----------
+        parameter_id (str)  :   id to save this parameter as
+        value (float)       :   initial value for this parameter
+        minimum (float)     :   minimum value that this parameter can take
+        maximum (float)     :   maximum value that this parameter can take
+        scale (float)       :   scaling when converting between [0,1] range
+        """
+        if parameter_id in self.parameters:
+            raise ValueError("parameter_id: {} already used".format(parameter_id))
+
+        self.parameters[parameter_id] = Parameter(value, minimum, maximum,
+                                                  scale, parameter_id)
+
+    def connect_parameter(self, parameter_id: str, module: SynthModule,
```

Comment:
I think we can get rid of `connect_parameter` given the strategy I'm describing below.

Reply:
I also find this pattern hard to grok, so am definitely looking for a way to avoid it :)

Reply:
I like this idea too. I can't quite picture it -- @jorshi do you have enough of a handle to implement it?

Reply:
Sweet, ya I think I've got an idea of how to implement that.